arelle/FileSource.py

```diff
@@ -342,6 +342,9 @@ def taxonomyPackageMetadataFiles(self):
         for f in (self.dir or []):
             if f.endswith("/META-INF/taxonomyPackage.xml"): # must be in a sub directory in the zip
                 return [f]  # standard package
+        # If not found in subdirectory, start by checking if META-INF is at archive top-level (and contains target file)
+        if "META-INF/taxonomyPackage.xml" in self.dir:
```

Comment:
Does removing the leading `/` from the existing `endswith(...)` check have the same effect?

```python
if f.endswith("META-INF/taxonomyPackage.xml")
    return [f]
```

Reply:
Also, it's worth mentioning that the taxonomy package spec requires a single top-level directory containing the `META-INF` sub directory.

[Taxonomy Package](https://www.xbrl.org/Specification/taxonomy-package/REC-2016-04-19/taxonomy-package-REC-2016-04-19.html#taxonomy-package)
> A [Taxonomy Package](https://www.xbrl.org/Specification/taxonomy-package/REC-2016-04-19/taxonomy-package-REC-2016-04-19.html#term-taxonomy-package) MUST contain a single top-level directory, with all other files being contained within that directory or descendant subdirectories (tpe:invalidDirectoryStructure). It is recommended, but not required, that the name of the top-level directory be the same as the name of the Taxonomy Package with the .zip file extension removed.
> 
> The top-level directory MUST contain a sub directory named META-INF (tpe:metadataDirectoryNotFound), which in turn MUST contain a taxonomyPackage.xml file (tpe:metadataFileNotFound). The META-INF directory MAY contain a catalog.xml file (see[ Section 3.3](https://www.xbrl.org/Specification/taxonomy-package/REC-2016-04-19/taxonomy-package-REC-2016-04-19.html#sec-remappings)). The META-INF directory MAY also contain other files, but no syntax or semantics are prescribed for such files by this specification.

Reply:
@ChristopheDeBernardi Can you address @austinmatherne-wk questions, please?

Reply:
[Sorry I did previously answer but as part as a "review" that I left pending, I'm not really used to writing PR.]

Removing the leading `/` would have the same end result, locating the `taxonomyPackage.xml` file (assuming there is only one of them).  
However preserving the existing loop before searching in other locations allows to:

* Explicitly look only in the expected location at first, then falling back to a "best effort" search (in the unlikely case where there would be more than one `taxonomyPackage.xml` file);
* log a warning or do another specific processing if the file exists but is misplaced.

Regarding the specification, it also says:

> A [Conformant Processor](https://www.xbrl.org/Specification/taxonomy-package/REC-2016-04-19/taxonomy-package-REC-2016-04-19.html#term-conformant-processor) MUST refuse to open or otherwise use a [Taxonomy Package](https://www.xbrl.org/Specification/taxonomy-package/REC-2016-04-19/taxonomy-package-REC-2016-04-19.html#term-taxonomy-package) that violates any of the "MUST" and "MUST NOT" statements in this document defined for Taxonomy Packages (...)

So, indeed, I wonder if looking for the file in alternate places is a good option. My main concern in this PR is that the method ends up returning the `catalog.xml` file if the `taxonomyPackage.xml` is not found.

On the other hand, I suspect that in the case of ESEF reports for instance some authorities would prefer a software to accept opening a package with a misplaced metadata file and see a warning rather than blocking the whole validation process because of that kind of irregularity.  
But that consideration goes beyond what I intended to do with this PR.